modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java

```diff
@@ -0,0 +1,426 @@
+                .contentType(ContentType.JSON)
+                .body(body).when().post(endpointURL);
+        response.then()
+                .log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_CREATED);
+
+        Optional<ApplicationListItem> b2bSelfServiceApp = oAuth2RestClient.getAllApplications().getApplications()
+                .stream().filter(application -> application.getName().equals("b2b-self-service-app")).findAny();
+        Assert.assertTrue(b2bSelfServiceApp.isPresent(), "B2B self service application is not created");
+        selfServiceAppId = b2bSelfServiceApp.get().getId();
+
+        JSONObject jsonObject = new JSONObject(readResource("organization-self-service-apis.json"));
+
+        for (Iterator<String> apiNameIterator = jsonObject.keys(); apiNameIterator.hasNext(); ) {
+            String apiName = apiNameIterator.next();
+            Object requiredScopes = jsonObject.get(apiName);
+
+            Response aPIResource =
+                    given().auth().preemptive().basic(authenticatingUserName, authenticatingCredential).when()
+                            .queryParam("filter", "identifier eq " + apiName).get("api-resources");
+            aPIResource.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK);
+            String apiUUID = aPIResource.getBody().jsonPath().getString("apiResources[0].id");
+
+            JSONObject authorizedAPIRequestBody = new JSONObject();
+            authorizedAPIRequestBody.put("id", apiUUID);
+            authorizedAPIRequestBody.put("policyIdentifier", "RBAC");
+            authorizedAPIRequestBody.put("scopes", requiredScopes);
+
+            Response authorizedAPIResponse =
+                    given().auth().preemptive().basic(authenticatingUserName, authenticatingCredential)
+                            .contentType(ContentType.JSON).body(authorizedAPIRequestBody.toString()).when()
+                            .post("applications/" + selfServiceAppId + "/authorized-apis");
+            authorizedAPIResponse.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK);
+        }
+    }
+
+    @Test(dependsOnMethods = "createApplicationForSelfOrganizationOnboardService")
+    public void getM2MAccessToken() throws Exception {
+
+        OpenIDConnectConfiguration openIDConnectConfiguration = oAuth2RestClient
+                .getOIDCInboundDetails(selfServiceAppId);
+        selfServiceAppClientId = openIDConnectConfiguration.getClientId();
+        selfServiceAppClientSecret = openIDConnectConfiguration.getClientSecret();
+        AuthorizationGrant clientCredentialsGrant = new ClientCredentialsGrant();
+        ClientID clientID = new ClientID(selfServiceAppClientId);
+        Secret clientSecret = new Secret(selfServiceAppClientSecret);
+        ClientAuthentication clientAuth = new ClientSecretBasic(clientID, clientSecret);
+        Scope scope = new Scope("SYSTEM");
```

Comment:
Shall we request the specific set of scopes here instead of SYSTEM

Reply:
This is resolved. Since the organization's self-service APIs are authorized in the `createApplicationForSelfOrganizationOnboardService()`, we decided to keep the scope as `'SYSTEM'`.